Lib/test/test_os.py

```diff
@@ -3328,40 +3238,52 @@ class TestSendfile(unittest.TestCase):
 
     @classmethod
     def setUpClass(cls):
-        cls.key = threading_helper.threading_setup()
         create_file(os_helper.TESTFN, cls.DATA)
 
     @classmethod
     def tearDownClass(cls):
-        threading_helper.threading_cleanup(*cls.key)
         os_helper.unlink(os_helper.TESTFN)
 
-    def setUp(self):
-        self.server = SendfileTestServer((socket_helper.HOST, 0))
-        self.server.start()
+    @staticmethod
+    async def chunks(reader):
+        while not reader.at_eof():
+            yield await reader.read()
+
+    async def handle_new_client(self, reader, writer):
+        self.server_buffer = b''.join([x async for x in self.chunks(reader)])
+        writer.close()
+        self.server.close()  # The test server processes a single client only
+
+    async def asyncSetUp(self):
+        self.server_buffer = b''
+        self.server = await asyncio.start_server(self.handle_new_client,
+                                                 socket_helper.HOSTv4)
         self.client = socket.socket()
-        self.client.connect((self.server.host, self.server.port))
-        self.client.settimeout(1)
-        # synchronize by waiting for "220 ready" response
-        self.client.recv(1024)
+        self.client.setblocking(False)
+        l = asyncio.get_running_loop()
+        await l.sock_connect(self.client, self.server.sockets[0].getsockname())
```

Comment:
Don't hesitate to name a loop as the loop :)
```suggestion
        loop = asyncio.get_running_loop()
        await loop.sock_connect(self.client, self.server.sockets[0].getsockname())
```

Reply:
The short name allowed to keep a 79 character limit. Though, on second thoughts, I took out a server address instead:

```diff
-        l = asyncio.get_running_loop()
-        await l.sock_connect(self.client, self.server.sockets[0].getsockname())
+        server_name = self.server.sockets[0].getsockname()
+        await asyncio.get_running_loop().sock_connect(self.client, server_name) 
```

Reply:
On *the third thoughts*, I moved `server_name = self.server.sockets[0].getsockname()` right under server creation. Without it, reading code as a prose becomes harder:

> Start the server. For a client, create a socket, set it to be nonblocking, get a server name that is a name of its listening port, connect to the server.

instead of:

> Start the server, its name is a name of its listening port. For a client, create a socket, set it to be nonblocking, connect to the server.